### PR TITLE
[Dashboard] Add soft-delete for cluster and job history rows

### LINKED ISSUE #8933: [Dashboard] Delete cluster / jobs from the history
We can just mark a row to be deleted, but not actually deleting it from the table

### PR DESCRIPTION
**Pull request by ambicuity**
## What changes are proposed in this PR?

Implements the ability to hide (soft-delete) cluster history and managed job entries from the dashboard, as described in #8933.

### Implementation

Rows are marked as dismissed using client-side `localStorage` — **no backend or database changes**. This aligns with the issue description: *"We can just mark a row to be deleted, but not actually deleting it from the table."*

**New files:**
- `sky/dashboard/src/hooks/useDismissedItems.js` — Reusable React hook for managing dismissed item IDs via `localStorage`
- `sky/dashboard/src/hooks/useDismissedItems.test.js` — 12 unit tests covering dismiss/restore/filter/persistence/edge cases

**Modified files:**
- `sky/dashboard/src/components/clusters.jsx` — Trash icon dismiss button for historical/terminated clusters + "Show N hidden" restore link
- `sky/dashboard/src/components/jobs.jsx` — Trash icon dismiss button for finished managed jobs + "Show N hidden" restore link

### User experience
- A 🗑️ (Trash2) icon appears in the Actions/Logs column for terminated/historical clusters and finished jobs
- Clicking it immediately hides the row from the dashboard
- A "Show N hidden clusters/jobs" link appears in the pagination area to restore all hidden items
- Dismissed state persists across page refreshes via `localStorage`
- Only finished/terminated items can be dismissed (not active ones)

## Tested:
- All 12 new unit tests pass (`jest src/hooks/useDismissedItems.test.js`)
- Full test suite: All existing tests still pass (24 passed; 6 pre-existing failures in version-display unrelated to this PR)
- ESLint: ✅ No warnings or errors
- Next.js production build: ✅ Compiles successfully

Fixes #8933

### CODE REVIEW

File: sky/dashboard/src/hooks/useDismissedItems.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+/**
```
Comment (Michaelvll) [inline sky/dashboard/src/hooks/useDismissedItems.js:1]: Seems we also need to mark the soft deleted rows in backend? Otherwise, it the state won't persist when the browser state clears or using a different device?
